k8s-charm/lib/charms/juju_dashboard/v0/juju_dashboard.py

```diff
@@ -78,10 +80,12 @@ def __init__(self, data):
         """
         # FIXME: Quick hack to fix a k8s bug in the controller charm.
         controller_url = data.get("controller-url")
-        controller_url = controller_url.replace("[", "").replace("]", "")
+        controller_url = (
+            controller_url.replace("[", "").replace(":0]", "").replace("]", "")
```

Comment:
In my case (using k3s instead of microk8s) the controller URL value is `wss://[10.43.67.21:0]:17070/api`

Reply:
This feels pretty hacky (both before and after). Why are we not receiving a correctly formatted controller url? Is this an issue with the operator framework and/or with the controller charm?

Reply:
Penny mentioned in the comment that this is sent by the Controller charm so I'm not sure where exactly this value is defined.

Reply:
FYI @barrettj12, the `controller_url` value is taken from the event relate's data field.

This quick hack isn't clean at all but is a quick work around that won't harm the charm in case the value is set properly.

Reply:
Sure, but my point is: we should fix the bug in the controller charm rather than putting workarounds on the other side. If you have time, you could look at https://github.com/juju/juju-controller and see if you can find the bug.

Reply:
Thanks for bringing this up! I totally agree that fixing the bug in the controller charm would be the best solution. I'm still learning about how charms work, so I'm not sure if I'll be able to find the bug right away. But I'll definitely take a look at the juju-controller repository and see if I can help out in any way. Thanks for the suggestion!

Reply:
tracked this bug down, it's actually a Juju issue rather than a controller charm issue. Juju sets this value in the controller charm config. So the fix will be in Juju. Probably we should deprecate the `controller-url` key and have a `controller-ip` / `controller-port` instead.